modules/programs/ghostty.nix

```
@@ -13,7 +13,11 @@ in {
   options.programs.ghostty = {
     enable = lib.mkEnableOption "Ghostty";
 
-    package = lib.mkPackageOption pkgs "ghostty" { };
+    package = lib.mkPackageOption pkgs "ghostty" {
+      nullable = true;
+      extraDescription =
+        "On darwin set this to null to manage ghostty config while ghostty is not present in nixpkgs.";
+    };
```

Just to capitalize ghostty in the description
```suggestion
    package = lib.mkPackageOption pkgs "Ghostty" { default = [ "ghostty" ]; };
```

---

That doesn't appear to work when nixpkgs is marking the package as [broken](https://github.com/NixOS/nixpkgs/blob/d5d576db277dbfb1463b47d77fddba818eff69f7/pkgs/by-name/gh/ghostty/package.nix#L199). Setting the package to null in config isn't allowed (not nullable) and nixpkgs evaluates the default since it _is_ present. This does work when the package isn't present at all.

---

It won't be present look at: https://github.com/nix-community/home-manager/pull/6300?notification_referrer_id=NT_kwDOAqUkqrQxNDIwOTgxOTU1Mjo0NDM3NzI1OA#discussion_r1910173889

---

Testing.

---

Yup, my mistake on where the actual eval was coming from. "Ghostty" isn't needed if we're guarding all the package sites. `mkPackageOption` currently only checks for presence not whether or not the package is broken.

I considered checking if the package is broken instead and looked at [~hostPlatform](https://github.com/NixOS/nixpkgs/blob/c0b4b9614390535f61f103fff1e48255a588fa4b/lib/systems/inspect.nix#). I think Linux is the only supported output from `hostPlatform` so that doesn't seem to buy much generality.

---

Turns out this trying to avoid installing only on linux brings in different complications. When we just gate on `isLinux` we add a synchronization point with nixpkgs. If nixpkgs gains a darwin version of ghostty (as some are working on) this module will have to change at the ~same as it's introduced into nixpkgs. Addtionally, to a lesser degree, it also interferes with users ability to set their own package if they figure out some work around for themselves. If, say, a *bsd or darwin user gets a package working for themselves before it lands in nixpkgs they would not be able to use the module as is.

There's a bit of weirdness around a transition from broken -> working for macos users too. If a nixpkgs copy exists some users will want that copy and some will want the macos managed version instead. (technically this could also be the case for nix on Arch)

To me this suggests setting `nullable: true` and handling the null case. Trying to default to the 'right' value for darwin users runs into that point above around choosing darwin or nix hosted packages. My user facing suggestion for that is adding an assertion: ~"isBroken, isNull`-> "programs.ghostty.package is marked as broken on your platform, set the value to null to manage your configuration without installing the package"

---

Handling null for `package` is exactly the path I have chosen for my private Ghostty module. I support that route.